### PR TITLE
Update IErrorBase to extend Error and include getTelemetryProperties

### LINKED ISSUE #8210: Container's "closed" event does not document well PII considerations of logging error objects
We recommend consumers to use our APIs to monitor errors and log in there telemetry pipeline.
So using "closed" event, it compiles to IErrorBase interface.
But it's pretty barebones interface and it does not tell all possible combinations of properties there can be (and we can't).
But most important - it does not point to a fact that all properties hae to be of TelemetryEventPropertyType | ITaggedTelemetryPropertyType, and proper description how to think about properties that might have PII content.
That seems like a must, in a form of IErrorBase clearly articulating shape of all properties (including unknown properties).

A great addition on top of that would be a stringify function for errors that would be able to strip all potentially PII holding properties (similar how we do for loggers that do not understand that structure).

Putting it in November, as Teams folks (Marco) are hitting it right now. Some guidance is required with some quick follow up, but maybe we can do it in December.

### PR DESCRIPTION
**Pull request by markfields**
Fixes #8210

### CODE REVIEW

File: common/lib/container-definitions/src/error.ts [hunk 1]
```diff
@@ -36,14 +38,30 @@ export enum ContainerErrorType {
...
      * - runtime
      * - drivers
      */
     readonly errorType: string;
+
+    /**
+     * See Error.message
+     * Privacy Note - This is a freeform string that we may not control in all cases (e.g. a dependency throws an error)
+     * If there are known cases where this contains privacy-sensitive data it will be tagged and included in the result
+     * of getTelemetryProperties. When logging, consider fetching it that way rather than straight from this field.
```
Comment (markfields) [inline common/lib/container-definitions/src/error.ts:53]: See #6609 for additional thinking here.  I think this comment is sufficient, the concern is somewhat hypothetical.

File: common/lib/container-definitions/src/error.ts [hunk 2]
```diff
@@ -36,14 +38,30 @@ export enum ContainerErrorType {
 /**
  * Base interface for all errors and warnings at container level
  */
-export interface IErrorBase {
+export interface IErrorBase extends Partial<Error> {
```
Comment (vladsud) [inline common/lib/container-definitions/src/error.ts:41]: Given that we keep adding same things to IFluidErrorBase & IErrorBase, I have hard time reasoning why there needs to be two almost identical types that have no direct relationship to each other. Should at least one inherit from another and clearly spell out in comments when one should expect one over another?

Comment (vladsud) [inline common/lib/container-definitions/src/error.ts:41]: What's the value of Partial<Error> to consumers? It feels none, and the only benefit is ensuring that types for message/name/stack are compatible with Error definitions, right? I.e. it's more of an internal minimal help.
Maybe other way to ask - can it be required (non-partial)?

File: common/lib/container-definitions/src/error.ts (hunk 1 above)
Comment (vladsud) [reply common/lib/container-definitions/src/error.ts:53]: Can we generalize? I.e. say - all these fields (including stack) are only for local debugging. If you need to log anything to telemetry, you have to go through getTelemetryProperties() path and respect tags

File: common/lib/container-definitions/src/error.ts (hunk 2 above)
Comment (markfields) [reply common/lib/container-definitions/src/error.ts:41]: Yeah see #6980.

Comment (markfields) [reply common/lib/container-definitions/src/error.ts:41]: The breaking change validation stuff blew up when I made it required.  I would like it to be required, but felt unsure of the ramifications of doing so on account of that, so thought it was reasonable to start here.

Comment (markfields) [reply common/lib/container-definitions/src/error.ts:41]: I can revisit making it required in the next release, when I have time to think through the N/N-1 concerns

Comment (markfields) [reply common/lib/container-definitions/src/error.ts:41]: I am still pondering yours and Tony's feedback on fluidErrorCode etc.  For now I'm content to have IFluidErrorBase be our internal interface and IErrorBase be the public API.  I'm actively thinking about this though.

**Review by vladsud — APPROVED** [review]
I'd love to see follow ups here sooner than later :)

**Pull request merged**